source/features/patch-diff-links.tsx

```diff
@@ -1,37 +1,42 @@
 import React from 'dom-chef';
-import elementReady from 'element-ready';
 import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import {getCleanPathname} from '../github-helpers/index.js';
+import observe from '../helpers/selector-observer.js';
 
-async function init(): Promise<void> {
+async function addPatchDiffLinks(commitMeta: HTMLElement): Promise<void> {
 	let commitUrl = '/' + getCleanPathname();
 
 	// Avoids a redirection
 	if (pageDetect.isPRCommit()) {
 		commitUrl = commitUrl.replace(/\/pull\/\d+\/commits/, '/commit');
 	}
 
-	const commitMeta = await elementReady('.commit-meta');
 	commitMeta!.classList.remove('no-wrap'); // #5987
-	commitMeta!.lastElementChild!.append(
-		<span className="sha-block" data-turbo="false">
-			<a href={`${commitUrl}.patch`} className="sha">patch</a>
+	commitMeta!.append(
+		<span className="sha-block ml-2" data-turbo="false">
+			<a href={`${commitUrl}.patch`} className="sha color-fg-default">patch</a>
 			{' '}
-			<a href={`${commitUrl}.diff`} className="sha">diff</a>
+			<a href={`${commitUrl}.diff`} className="sha color-fg-default">diff</a>
 		</span>,
 	);
 }
 
+async function init(signal: AbortSignal): Promise<void> {
+	observe([
+		'.commit-meta', // TODO remove in March 2025
+		'pre:has([data-hovercard-url])',
```

Comment:
This is too generic unfortunately.

Also I think we should just mark the old selector as `isPRFiles` instead of dropping it. Can you ensure it still works since you dropped `lastElementChild` on  [`ef22049` (#7909)](https://github.com/refined-github/refined-github/pull/7909/commits/ef22049cf89d2934640ba350c16d3fd49b2ad97e)? Maybe it the new selector should be `.commit-meta > span:last-child`

Reply:
> Also I think we should just mark the old selector as `isPRFiles` instead of dropping it. Can you ensure it still works since you dropped `lastElementChild` on [`ef22049` (#7909)](https://github.com/refined-github/refined-github/pull/7909/commits/ef22049cf89d2934640ba350c16d3fd49b2ad97e)? Maybe it the new selector should be `.commit-meta > span:last-child`

It's ok.

Reply:
![CleanShot 2024-10-20 at 00 12 56@2x](https://github.com/user-attachments/assets/c783f968-7291-471e-bd9e-603f8f36bd95)

Reply:
> This is too generic unfortunately.

how about `pre:has([aria-label^="Copy full SHA for"])` ?

![CleanShot 2024-10-20 at 00 20 41@2x](https://github.com/user-attachments/assets/023a1ab0-74f5-4967-baa1-86b7f220181b)